QWC2Components/components/Search.jsx

```diff
@@ -304,25 +304,54 @@ class Search extends React.Component {
         );
     }
     showResult = (item, zoom=true) => {
-        if((item.type || SearchResultType.PLACE) === SearchResultType.PLACE) {
-            this.props.removeLayer("searchselection");
-            let text = item.label !== undefined ? item.label : item.text;
-            if(zoom) {
-                // find max zoom level greater than min scale
-                let maxZoom = 0;
-                const scales = this.props.map.scales;
-                for (let i in scales) {
-                    if (scales[i] < this.props.searchOptions.minScale) {
-                        break;
+        if(zoom) {
+            let bbox = item.bbox.slice(0) || [];
+            let crs = item.crs;
+            let x = item.x;
+            let y = item.y;
+
+            // find max zoom level greater than min scale
+            let maxZoom = 0;
+            const scales = this.props.map.scales;
+            for (let i in scales) {
+                if (scales[i] < this.props.searchOptions.minScale) {
+                    break;
+                } else {
+                    maxZoom = i;
+                }
+            }
+
+            if(SearchResultType.THEMELAYER && item.layer) {
```

Comment:
I suppose this should be

    if(item.type === SearchResultType.THEMELAYER && item.layer)

Reply:
yep! should be fixed in the latest commit